Revert "Fix EntityReference not finding the entity if the parent was not added to the scene yet. (#4081)"

jpauloruschel opened the pull request:
This reverts commit 33af0a817dae2db5f996f66d2c546f0218009897, PR https://github.com/playcanvas/engine/pull/4081

Fixes https://github.com/playcanvas/engine/issues/4114 

Note that the original issue that the reverted PR was fixing is also fixed by https://github.com/playcanvas/engine/pull/4054.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

slimbuck commented:
looks like this PR is changing more than just #4081?

willeastcott commented:
Ah yeah, I updated the `EntityReference` unit tests since the original PR was merged.

jpauloruschel commented:
> Ah yeah, I updated the `EntityReference` unit tests since the original PR was merged.

Yes, I updated the PR now to revert the logic, rather than the entire file changes.
